src/EventSubscriber/CacheableResponseSubscriber.php

```diff
@@ -90,13 +90,24 @@ public function onRespond(ResponseEvent $event) {
       }
 
       $tags_string = implode(' ', $tags);
-      if (25000 < strlen($tags_string)) {
-        $tags_string = substr($tags_string, 0, 25000);
-        // The string might have cut of in the middle of a tag.
-        // So now find the the last occurence of a space and cut to that length.
+      // value also defined in config install and schema files
+      $limit = 25000;
+
+      // Allow to lower the surrogate key header limit on non-Pantheon Env
+      if (!isset($_ENV['PANTHEON_INFRASTRUCTURE_ENVIRONMENT'])) {
+        $configured_limit = $this->configFactory->get('pantheon_advanced_page_cache.settings')->get('surrogate_key_header_limit');
+        // if no local config limit, default it to zero and does not exceed 25,000.
+        $limit = min((int) ($configured_limit ?? 0), $limit);
```

Comment:
This should default to 25000 if unconfigured. "Unlimited" exceeds 25000, and we don't want that to happen.